### ISSUE TITLE
[FEA] API method for removing empty partitions from a dataframe

### ISSUE THREAD

**Issue by beckernick** [dask/dask#5729]
It would be nice if there was a clean API method to remove empty partitions from a dataframe. This is particularly relevant given issues like #5726 .

**Comment by mrocklin** [comment]
Most dask dataframe operations respect laziness.  This wouldn't and so might surprise users.  

As an alternative, there has been some effort to make things like `df = df.repartition(bytes='1 GB')` work.  This also has the same concern that it forces the computation of df (there is no way to see how big partitions are otherwise), but at least it's only one such case that is more generally useful, rather than having many small utility functions, some of which do or do not trigger computation.

**Comment by mrocklin** [comment]
Some version of this may already be implemented here: https://github.com/dask/dask/pull/4416

**Comment by beckernick** [comment]
Interesting, thanks for the link. I'll take a look down that thread.

To your comment on surprising users, I think in general you're right. But, I wonder if users who want to actively remove empty partitions may be more aware of the implications. Perhaps a common moment that a user notices empty partitions is while debugging downstream failures (extrapolating, perhaps wrongly, from myself). In that case, an API method to achieve this (perhaps with good documentation or a warning) would be appealing,

**Comment by mrocklin** [comment]
> But, I wonder if users who want to actively remove empty partitions may be more aware of the implications

I'm sure that they do, but other users will come by and they'll see the `df.remove_empty_partitions()` method and they'll think "well sure, that's a great idea".  

Anyway, I suspect that you have what you want already in the `repartition` method.

**Comment by jrbourbeau** [comment]
I agree with @mrocklin that we generally want to keep dataframe operations lazy. @beckernick did using `repartition` end up working for your use case?
